File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program175.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a java program which accept string from user and display it in reverse order.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class StringDemo
{
	public void Reverse(String str)
	{
		int iCnt=str.length()-1,iCount=0;
		while(iCnt!=-1)
		{
			System.out.print(str.charAt(iCnt));
			iCnt--;
		}

		System.out.println();
		
	}
}


class Program
{
	public static void main(String arg[])
	{
	
		Scanner sObj = new Scanner(System.in);
		System.out.println("Input: ");
		String str = sObj.nextLine();
	
		StringDemo sobj = new StringDemo();
		sobj.Reverse(str);
	
	}
}

/*
OUTPUT:

Input:
Vishal
lahsiV

*/

File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program175.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a java program which accept string from user and display it in reverse order.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
#include<stdio.h>

using namespace std;

class StringInput
{
	public:

		char Arr[30];

		void Accept()
		{
			cout<<"Enter the String: ";
			scanf("%[^'\n']s",Arr);
		}

};

class StringDemo:public StringInput
{
	public:

		void Reverse()
		{
			char *str = Arr;
			int iCnt=0;
			char Temp;

			while(*str!='\0')
			{
				iCnt++;
				str++;
			}
			
			int iEnd=iCnt-1;
			iCnt=0;
			while(iCnt<iEnd)
			{
				Temp = Arr[iCnt];
				Arr[iCnt] = Arr[iEnd];
				Arr[iEnd] = Temp;

				iCnt++;
				iEnd--;
			}

			cout<<"Reverse is: "<<Arr<<"\n";
		}
};

int main()
{

	StringDemo *sObj = new StringDemo();
	sObj->Accept();	
	sObj->Reverse();
}

/*
output:

Enter the String: Vishal
Reverse is: lahsiV
*/
